fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/TypeRdfContext.java

```diff
@@ -58,71 +61,36 @@ public TypeRdfContext(final FedoraResource resource,
 
         //include rdf:type for primaryType, mixins, and their supertypes
         concatRdfTypes();
+        if (resource() instanceof NonRdfSource) {
+            // gather versionability info from the parent
+            if (resource().getNode().getParent().isNodeType("mix:versionable")) {
```

Comment:
Instead of relying on `resource().getNode().getParent()`, it may be more maintainable if you use: 
`((NonRdfSource) resource()).getDescription().getNode().isNodeType(...)`
...which raises the question of versionability of NonRdfSource properties. Do we care if the description is versionable if in fact the properties (stored on the NonRdfSource) are not themselves versionable?